synapse/storage/data_stores/main/events.py

```diff
@@ -1714,10 +1717,24 @@ def _purge_room_txn(self, txn, room_id):
         #       index on them. In any case we should be clearing out 'stream' tables
         #       periodically anyway (#5888)
 
+        # Now we fetch all the state groups that should be deleted.
+        txn.execute(
+            """
+                SELECT DISTINCT state_group FROM events
```

Comment:
erm, haven't we just deleted all the relevant rows from `events`?

Reply:
Gah. And the tests were broken.